Sunflower (#546)
folio-org/docs: content/en/docs/Platform essentials/Permissions/_index.md

	- Cannot be nested. Roles are flat
	- Are reusable 

- Default roles
	- Roles provided by the system, optionally loaded during tenant entitlement 

- Policies 
	- Provides greater control over who can do what, when, and from where. For example, policies could be created to only allow users to perform check-in/check-out operations during normal business hours from certain locations
	- Policies are associated with capabilities 
	- Time-based policies are managed by administrators, not by the system 

- AuthUser
	- Refers to a user record in [Keycloak](https://github.com/folio-org/folio-keycloak)
	- AuthUsers are managed by FOLIO. No direct interaction with Keycloak is required
	- AuthUsers are a subset of Users, which include staff, patrons, administrators, etc. 
	- A user becomes an AuthUser if they are assigned credentials, roles, or capabilities. The Keycloak user record is only needed for those who log in and use FOLIO. There is no need for Keycloak to know about patrons and others who are just data in the system 

**How do these relate to one another?** 
This diagram may help you visualize the relationships between some of these terms: 

> ![image](/relation-diagram.png)

Reviewer: ```suggestion
![image](relation-diagram.png)
```

![image](role-creation-2.png)

![image](role-creation-3.png)

After selecting one or more applications and clicking "**Save and close**," the Capability and Capability Set portions of the role creation form will be populated, and you can select those which you want to include in your role by checking individual boxes or the boxes in the column headers. 

![image](role-creation-4.png)

#### Notes on selecting Capabilities and Capability Sets: 
* Capabilities and Capability Sets are divided into 3 groups: **Data**, **Settings**, and **Procedural**. These are intended to make it easier to sort through the options. Here are brief descriptions of each set: 
	*<u>Data</u>: Capablities for directly managing resources as they exist in FOLIO (i.e. RESTful APIs)
		*E.g. "Instance," "purchase order lines," etc. 
	*<u>Procedural</u>: Capabilities for initiating and controlling processes in FOLIO (i.e. execution of tasks)
		*E.g. "Check-out-by-barcode" 
	*<u>Settings</u>: Administrative Capabilities for managing FOLIO configurations 

> 		*E.g. "Notes settings" 

Reviewer: ```suggestion
	* **Data**: Capablities for directly managing resources as they exist in FOLIO (i.e. RESTful APIs)
		*E.g. "Instance," "purchase order lines," etc. 
	* **Procedural**: Capabilities for initiating and controlling processes in FOLIO (i.e. execution of tasks)
		*E.g. "Check-out-by-barcode" 
	* **Settings**: Administrative Capabilities for managing FOLIO configurations 
		*E.g. "Notes settings" 
```

> 		*E.g. "Notes settings" 

Reviewer: `<u>` is no supported by Markdown. You may use `*single*`, `**double**` or `***triple***` asterisks instead: https://spec.commonmark.org/0.31.2/#emphasis-and-strong-emphasis

![image](users-6.png)

***N.B.*** Don't forget to click "Save & close" on the user edit form to save your changes. If you cancel now, your changes will not take effect. 

In some cases you may see a dialgoue asking you to confirm the creation of a user record in Keycloak. The creation of this record is necessary for the role assignment to succeed. Click "**Confirm**" to proceed, or "**Cancel**" if you're unsure. 

![image](users-7.png)

Upon completion, you will land back at the user detail view. The "**User roles**" accordion shouild be updated to reflect the changes you've just made. 

## Migration
When migrating from the Okapi platform to Eureka adjustments must be made. Some of this will be handled by migration APIs. Other parts will need to be performed by a human.  

### Migration APIs
There are two APIs which do most of the heavy lifting. The goal of these is to ensure users who could perform certain actions on Okapi, can still perform those same actions on Eureka.
* <u>Users migration</u> - provided by mod-users-keycloak. This API will create AuthUser for any User in the system which has at least one permission assigned to them

> * <u>Roles migration</u> - provided by mod-roles-keycloak. This API looks at the permissions assigned to users, and creates a Role for each unique set. These system-generated roles are then assigned to the appropriate users

Reviewer: ```suggestion
* **Users migration** - provided by mod-users-keycloak. This API will create AuthUser for any User in the system which has at least one permission assigned to them
* **Roles migration** - provided by mod-roles-keycloak. This API looks at the permissions assigned to users, and creates a Role for each unique set. These system-generated roles are then assigned to the appropriate users
```

> * <u>Roles migration</u> - provided by mod-roles-keycloak. This API looks at the permissions assigned to users, and creates a Role for each unique set. These system-generated roles are then assigned to the appropriate users

Reviewer: `<u>` is no supported by Markdown. You may use `*single*`, `**double**` or `***triple***` asterisks instead: https://spec.commonmark.org/0.31.2/#emphasis-and-strong-emphasis
